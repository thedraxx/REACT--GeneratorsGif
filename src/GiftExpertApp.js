import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export function GiftExpertApp({defaultCategories = []}) 
{
    const [categories, setCategories] = useState(defaultCategories);
    return (
        <>
            <h2 className ="tittle">GIF GENERATOR</h2>
            <AddCategory setCategories = {setCategories}/>

            <hr />
            <ol>
                
                {  
                    categories.map( (category) => (
                        <GifGrid 
                            key = {category}
                            category={category} 
                        />
                    ))
                }
            </ol>
        </>
    )
}

