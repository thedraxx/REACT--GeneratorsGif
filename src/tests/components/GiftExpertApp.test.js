import React from 'react';
import { shallow } from 'enzyme';
import { GiftExpertApp } from '../../GiftExpertApp';

describe('Pruebas en <GiftExpertApp />', () => {
    
    test('Hacer un snapShoot', () => {
        
        const wrapper = shallow(<GiftExpertApp />);
        expect (wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['one punch','dragon Ball'];
        const wrapper = shallow(<GiftExpertApp defaultCategories = {categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);



    })
    
    

})
