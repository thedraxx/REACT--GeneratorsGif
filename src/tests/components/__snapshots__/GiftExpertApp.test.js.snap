// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2
    className="tittle"
  >
    GIF GENERATOR
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon Ball"
      key="dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GiftExpertApp /> Hacer un snapShoot 1`] = `
<Fragment>
  <h2
    className="tittle"
  >
    GIF GENERATOR
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
